x-pack/plugins/reporting/server/routes/common/jobs/get_document_payload.ts

```diff
@@ -84,29 +87,29 @@ export function getDocumentPayloadFactory(reporting: ReportingCore) {
     };
   }
 
-  // @TODO: These should be semantic HTTP codes as 500/503's indicate
-  // error then these are really operating properly.
   async function getFailure({ id }: ReportApiJSON): Promise<Payload> {
-    const jobsQuery = jobsQueryFactory(reporting);
+    const jobsQuery = jobsQueryFactory(reporting, { isInternal });
     const error = await jobsQuery.getError(id);
 
-    logger.debug(`Report job ${id} has failed. Sending statusCode: 500`);
+    // For download requested over public API, status code for failed job must be 500 to integrate with Watcher
+    const statusCode = isInternal ? 200 : 500;
```

Comment:
Was just curious if 4** is a better status instead of 200.  4** shouldn't be noisy in the logs

Reply:
@Dosant I did some resource on status codes, and found that **202 Accepted** is intended for this scenario: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202

> The HyperText Transfer Protocol (HTTP) 202 Accepted response status code indicates that the request has been accepted for processing, but the processing has not been completed; in fact, processing may not have started yet. The request might or might not eventually be acted upon, as it might be disallowed when processing actually takes place.

Another option could be to use **403 Forbidden** but after looking more into it, **202** seems like a better option. Do you agree?

Reply:
Resolved via 39f2267ac87daf6ff75edecddd7026e9f12fe9a6